Src/FluentAssertions/AssertionExtensions.cs

```diff
@@ -917,7 +917,31 @@ public static TaskCompletionSourceAssertions<T> Should<T>(this TaskCompletionSou
     /// <exception cref="ArgumentNullException"><paramref name="eventSource"/> is <see langword="null"/>.</exception>
     public static IMonitor<T> Monitor<T>(this T eventSource, Func<DateTime> utcNow = null)
     {
-        return new EventMonitor<T>(eventSource, utcNow ?? (() => DateTime.UtcNow));
+        var options = new EventMonitorOptions();
+
+        if (utcNow is not null)
+        {
+            options.ConfigureTimestampProvider(utcNow);
+        }
+
+        return new EventMonitor<T>(eventSource, options);
+    }
```

Comment:
Now that we're open to change the existing API, we could remove `Func<DateTime> utcNow = null`, such that this becomes.
```cs
public static IMonitor<T> Monitor<T>(this T eventSource) =>
    new EventMonitor<T>(eventSource, new EventMonitorOptions());
```

For testing we still have the internal `ConfigureTimestampProvider`.

@dennisdoomen What do you think?

Reply:
> For testing we still have the internal `ConfigureTimestampProvider`.

What's that?

Reply:
`EventMonitorOptions.ConfigureTimestampProvider`.

In #625 the `Func<DateTime> utcNow = null` was added, but I suspect it wasn't meant for public use, but merely a necessity for testing.

Reply:
Ah yes, you're right. That would no longer be necessary.

Reply:
Do you have a workflow for obsolete methods? Should it be tagged obsolete first with the remart it will be removed in the next version? Then people have some time, if they are using it for some reason.

Reply:
Yes, that's how we normally do it. But last week, the `develop` branch is used for the next major version, so we're fine to introduce well-argumented breaking changes in that branch.

Reply:
I removed the utcnow. I handed down the options to the eventrecorder, as otherwise the late changed reference on the timestamp provider would not be used. The events are subscribed when the constructor is called. Please check if it is conform to your other code and let me know, if it you like another approach.

Reply:
> In https://github.com/fluentassertions/fluentassertions/pull/625 the Func<DateTime> utcNow = null was added, but I suspect it wasn't meant for public use, but merely a necessity for testing.

We should mention this in the docs as well :)